File: Dante1/Module1.08/Zadanie8.01/main.c
#include <stdio.h>

unsigned long factorial_rec(int n);

int main()
{
    int num;

    printf("Enter: ");

    int check = scanf("%i",&num);

    if(check != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    if(num < 0 || num > 20)
    {
        printf("Incorrect input data");
        return 2;
    }

    printf("%lu", factorial_rec(num));

    return 0;
}

unsigned long factorial_rec(int n)
{
    if(n < 0)
    {
        return 0;
    }
    if(n <= 1)
    {
        return 1;
    }
    return n * factorial_rec(n - 1);
}

File: Dante1/Module1.08/Zadanie8.01/main.rs
fn factorial_rec(num: i32) -> u64
{
    if num < 0
    {
        return 0;
    }
    else if num <= 1
    {
        //println!("dupa {}", num);
        return 1;
    }
    //println!("Num is: {}", num);
    (num as u64) * factorial_rec(num - 1)

}

fn main()
{
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let num: i32 = input.trim().parse().expect("Incorrect input");

    if 0 > num || num > 20
    {
        panic!("Too much");
    }

    println!("Result is: {}", factorial_rec(num));
}
